### PR TITLE
feat: update survey for the edi project

### PR DESCRIPTION
**Pull request by NuclearRedeye**

### CODE REVIEW

File: public/locales/en/survey.json [hunk 1]
```diff
@@ -1,12 +1,321 @@
...
+            { "value": "UA", "label": "Ukraine" },
+            { "value": "AE", "label": "United Arab Emirates" },
+            { "value": "GB", "label": "United Kingdom" },
+            { "value": "US", "label": "United States" },
+            { "value": "UM", "label": "United States Minor Outlying Islands" },
+            { "value": "UY", "label": "Uruguay" },
+            { "value": "UZ", "label": "Uzbekistan" },
+            { "value": "VU", "label": "Vanuatu" },
+            { "value": "VE", "label": "Venezuela, Bolivarian Republic of" },
+            { "value": "VN", "label": "Viet Nam" },
```
Comment (Cooryd) [inline public/locales/en/survey.json:290]: Is this correct 🤔

File: src/survey/styles/survey.scss [hunk 2]
```diff
@@ -1,3 +1,43 @@
 .survey {
   @include standardGrid;
+
+  h2 {
+    margin-bottom: 27px;
```
Comment (Cooryd) [inline src/survey/styles/survey.scss:5]: We'll need to convert these to `rem` values 🔢

File: src/survey/components/SurveyPart2.tsx [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+}
+
+interface Props {
+    id?: string;
+    next?: (answers: SurveyPageAnswers) => void;
+    previous?: () => void;
+    defaultValues?: SurveyPageAnswers;
+}
+
+const countryCodes = [
```
Comment (NuclearRedeye) [inline src/survey/components/SurveyPart2.tsx:24]: This might be better placed in a separate file

File: src/ui/atoms/Button.tsx [hunk 4]
```diff
@@ -12,9 +13,11 @@ const Button: React.FC<Props> = ({
     className,
     disabled = false,
     loading = false,
+    isSubmit = false,
     ...rest
 }: Props): JSX.Element => (
     <button
+        type={isSubmit ? 'submit' : undefined}
```
Comment (Cooryd) [inline src/ui/atoms/Button.tsx:20]: Lets add a unit test for this prop

File: tests/page-objects/SurveyPage.ts [hunk 5]
```diff
@@ -2,52 +2,121 @@ import { Selector, t } from 'testcafe';
...
+    private async assertOnPage1(): Promise<void> {
+        // Container is present
+        await t.expect(this.page1.exists).ok();
+
+        // Visible fields are present
+        await t.expect(this.submittingAs.exists).ok();
+        await t.expect(this.independentResearcher.exists).ok();
+
+        // Hidden fields are not visible
+        await t.expect(this.independentResearcherYear.exists).notOk();
```
Comment (Cooryd) [inline tests/page-objects/SurveyPage.ts:37]: The function name `assertOnPage1` doesn't specify that the form is clear of any values. We might get caught out with this assertion if we were to add extended survey tests later for navigating forward then back to page 1. I think the unit tests cover the conditionality of the `independentResearcherYear` without the need for an additional check here.

File: tests/page-objects/SurveyPage.ts [hunk 6]
```diff
@@ -2,52 +2,121 @@ import { Selector, t } from 'testcafe';
...
+
+        // Visible fields are present
+        await t.expect(this.genderIdentity.exists).ok();
+        await t.expect(this.countryOfResidence.exists).ok();
+        await t.expect(this.secondCountryOfResidenceToggle.exists).ok();
+        await t.expect(this.countryIndentifyAs.exists).ok();  
+        await t.expect(this.raceOrEthnicity.exists).ok();
+
+        // Hidden fields are not visible
+        await t.expect(this.genderSelfDescribe.exists).notOk();
```
Comment (Cooryd) [inline tests/page-objects/SurveyPage.ts:52]: Same as above

File: tests/page-objects/SurveyPage.ts [hunk 7]
```diff
@@ -2,52 +2,121 @@ import { Selector, t } from 'testcafe';
...
     public async populateForm(): Promise<void> {
-        await this.answerQuestion(1);
-        await this.answerQuestion(2);
-        await this.answerQuestion(3);
+        await this.assertOnPage();
+        await this.populatePage1();
+        await t.expect(Selector(this.doneSkipButton).exists).ok();
+        await t.expect(Selector(this.doneSkipButton).visible).ok();
+        await clickSelector(this.doneSkipButton);
+        await t.wait(500);
```
Comment (Cooryd) [inline tests/page-objects/SurveyPage.ts:115]: The above 4 lines could probably do with being a private function

File: tests/page-objects/SurveyPage.ts [hunk 8]
```diff
@@ -2,52 +2,121 @@ import { Selector, t } from 'testcafe';
...
+        await this.populatePage2();
     }
 
-    public async skipOrFinish(): Promise<void> {
+    public async completeSurvey(): Promise<void> {
         await t.expect(Selector(this.doneSkipButton).exists).ok();
         await t.expect(Selector(this.doneSkipButton).visible).ok();
         await clickSelector(this.doneSkipButton);
```
Comment (Cooryd) [inline tests/page-objects/SurveyPage.ts:122]: These 3 lines could be replaced with a call to the private function suggested above

Comment (Cooryd) [reply tests/page-objects/SurveyPage.ts:122]: *4 lines, looks like the wait is below this comment

**Review by Cooryd — APPROVED** [review]
Approve pending unused var being removed

**Pull request merged**